pkg/driver/driver_aws.go

```diff
@@ -97,25 +97,32 @@ func (d *AWSDriver) Create() (string, string, error) {
 		return "Error", "Error", err
 	}
 
-	securityGroupIDs, err := d.generateSecurityGroups(d.AWSMachineClass.Spec.NetworkInterfaces[0].SecurityGroupIDs)
-	if err != nil {
-		return "Error", "Error", err
+	var networkInterfaceSpecs []*ec2.InstanceNetworkInterfaceSpecification
+	for i, netIf := range d.AWSMachineClass.Spec.NetworkInterfaces {
+		spec := &ec2.InstanceNetworkInterfaceSpecification{
```

Comment:
If I understand it correct, the only essential input here is the SecurityGroups and SubnetIDs. Do we really need other fields of `DeviceIndex`, `AssociatePublicIpAddress`, `DeleteOnTermination` and `Description` ? We can simply avoid setting them altogether otherwise.
Introducing these fields with golang-defaults could imply breaking API changes, and possibly unintentional behviour. :) 
- If you see these fields are useful, can we please follow a pattern where we set them only if they are specified in the MachineClass. We will have to convert these fields to a pointer, eg `bool` to `*bool` in that case.
- Please check the pattern followed here for [ref](https://github.com/gardener/machine-controller-manager/blob/master/pkg/driver/driver_aws.go#L180-L195) for DeleteOnTermination.

Reply:
The same as https://github.com/gardener/machine-controller-manager/pull/441#discussion_r407408752.


>`DeviceIndex`

It's required and not specified anywhere in the MachineClass. Simply an AWS API kink.

Reply:
Sure, it should be fine to keep it this way then, thanks for updating the types.